src/pages/BoardPage.tsx

```diff
@@ -11,43 +11,41 @@ import { CreateTaskModal } from "../components/boardPage/CreateTaskModal/CreateT
 import { GeneralErrorMessage } from "../components/boardPage/GeneralErrorMessage/GeneralErrorMessage";
 import { EditBoardModal } from "../components/boardPage/EditBoardModal/EditBoardModal";
 import { BoardSelectPage } from "./BoardSelectPage";
+import { useRecoilState } from "recoil";
+import { selectedBoardIdState } from "../appState/selectedBoardIdState";
 
 function BoardPage() {
-  const [boardId, setBoardId] = useState<number | null>(null);
+  const [selectedBoard, setSelectedId] = useRecoilState(selectedBoardIdState);
   const [projectImage] = useState("https://via.placeholder.com/150");
   const { board, fetchBoard } = useBoardState();
   const { fetchTaskLists, taskLists } = useTaskListState();
   const [error, setError] = useState<boolean>(false);
 
   async function initData() {
     try {
-      if (!boardId) {
+      if (!selectedBoard.id) {
         return;
       }
-      await fetchBoard(+boardId);
-      await fetchTaskLists(+boardId);
+      await fetchBoard(+selectedBoard.id);
+      await fetchTaskLists(+selectedBoard.id);
     } catch (error) {
       setError(true);
     }
   }
 
-  function handleBoardSelection(idBoard: number) {
-    setBoardId(idBoard);
-  }
-
   useEffect(() => {
     initData();
-  }, [boardId]);
+  }, [selectedBoard.id]);
 
-  if (!boardId) {
-    return <BoardSelectPage onBoardSelection={handleBoardSelection} />;
+  if (!selectedBoard.id) {
+    return <BoardSelectPage />;
   }
 
   if (error) {
     return (
       <GeneralErrorMessage
         onSelectededBoardIdReset={() => {
-          setBoardId(null);
+          setSelectedId({ id: null });
```

Comment:
Remove this one from here. Call setRecoilState from the component GeneralErrorMessage.

Reply:
https://recoiljs.org/docs/api-reference/core/useSetRecoilState/